File: REST API/C-Sharp/CreateContact.cs
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Threading.Tasks;

public class Createcontact
{
    static HttpClient client = new HttpClient(); // One HttpClient instance be reused.
                                                 //HttpClient is used to make WebAPI Request from console application.

    static async Task Main(string[] args)
    {
        string domain = "YOUR_DOMAIN"; //Your bolddesk domain name.

        string apiPath = "/api/v1/contacts"; //Your API path.

        string apiKey = "YOUR_API_KEY"; //Your API key.

        client.BaseAddress = new Uri("https://" + domain); //Generate API path to call.

        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json")); //Header Value for Content type.

        client.DefaultRequestHeaders.Add("x-api-key", apiKey); //For Authentication access.

        try
        {
            //Post data to create contacts.
            var contact = new Contact
            {
                ContactName = "Albert",
                EmailId = "albert345@gmail.com",
                SecondaryEmailId = "albert123@gmail.com",
                ContactDisplayName = "Albert einsten",             //Sample Data's for Required fields..        
                ContactPhoneNo = "9562341875",
                ContactMobileNo = "6532148975",
                ContactAddress = "5th Lane, East America",
                ContactJobTitle = "scientist",
                TimeZoneId = 103,
                LanguageId = 1,
                ContactNotes = "Sample Note",
                ContactExternalReferenceId = "A237",
                ContactTag = "Sample_Tag",
                IsVerified = true
            };

            //Post data to create contact detail.
            var contactDetail = new ContactDetail
            {
                ContactDisplayName = "Shanmugam",
                LanguageId = 1,
                ContactName = "Shanmuga perumal",                  //Sample Data's for Required fields..
                EmailId = "Shanmugam567@gmail.com",
                SecondaryEmailId = "Shanmugam854@gmail.com",
                ContactPhoneNo = "8569374125",
                ContactMobileNo = "9563241875",
                ContactAddress = "6th floor, Aruna complex, North madras, Tamilnadu, india",
                ContactJobTitle = "Plummer",
                ContactGroup = [{ "id": 962, "accessScopeId": 2, "isPrimary": true}],
                CustomFields = new Dictionary<string, object?>()
                {
                    { "cf_custom_dropdown_1": 74 }, 
                    {"cf_engineer": 7465 }, 
                    {"cf_laction": 5704}
                };  
            };
            HttpResponseMessage response = await client.PostAsJsonAsync(apiPath, contactData).ConfigureAwait(false);//To send a POST request as an asynchronous operation to the specified Uri with the given value serialized as JSON.

            var responseData = await response.Content.ReadAsStringAsync().ConfigureAwait(false); //To read the response.

            response.EnsureSuccessStatusCode(); //For Confirms whether the response is success or Not.

            Console.WriteLine(responseData); //Return the response.
        }

        //To catch the error message.
        catch (Exception ex)
        {
            Console.WriteLine("ERROR: ");
            Console.WriteLine("Status Code: " + ex.StatusCode); //Returns the error status.
            Console.WriteLine(ex.Message);//Returns the error message.
            Console.WriteLine(ex.StackTrace);//Returns the stack at the time uncaught exception occurs.  
        }

    }
}

public class Contact
{
    public string ContactDisplayName { get; set; }

    public int LanguageId { get; set; }

    public string ContactName { get; set; }

    public string EmailId { get; set; }

    public string SecondaryEmailId { get; set; }

    public string ContactPhoneNo { get; set; }

    public string ContactMobileNo { get; set; }

    public string ContactAddress { get; set; }

    public string ContactJobTitle { get; set; }

    public int TimeZoneId { get; set; }

    public string ContactNotes { get; set; }

    public string ContactExternalReferenceId { get; set; }

    public string ContactTag { get; set; }

    public bool IsVerified { get; set; }

}

public class ContactDetail
{
    public string ContactDisplayName { get; set; }

    public int LanguageId { get; set; }

    public string ContactName { get; set; }

    public string EmailId { get; set; }

    public string SecondaryEmailId { get; set; }

    public string ContactPhoneNo { get; set; }

    public string ContactMobileNo { get; set; }

    public string ContactAddress { get; set; }

    public string ContactJobTitle { get; set; }

    public List<ContactGroupCollectionObject>? ContactGroup { get; set; }

    public Dictionary<string, object?>? CustomFields { get; set; }
}

public class ContactGroupCollectionObject
{
    public long id { get; set; }

    public int accessScopeId { get; set; }

    public bool isPrimary { get; set; }
}

File: REST-API-Django/bolddesk_client/management/commands/create_contact.py
from django.core.management.base import BaseCommand
import requests
import traceback


class Command(BaseCommand):
    help = 'Create a contact in BoldDesk via API (sample).'

    # Reuse a single Session instance (similar to HttpClient in C#)
    session = requests.Session()

    def add_arguments(self, parser):
        parser.add_argument('--domain', default='YOUR_DOMAIN', help='BoldDesk domain (e.g., yourcompany.bolddesk.com)')
        parser.add_argument('--api-key', dest='api_key', default='YOUR_API_KEY', help='API key')
        parser.add_argument('--name', dest='contact_name', default='James', help='Contact name')
        parser.add_argument('--email', dest='email', default='james@example.com', help='Email ID')
        parser.add_argument('--display-name', dest='display_name', default='Jade', help='Contact display name')

    def handle(self, *args, **options):
        domain = options['domain']
        api_path = '/api/v1/contacts'
        api_key = options['api_key']
        contact_name = options['contact_name']
        email = options['email']
        display_name = options['display_name']

        base_url = f'https://{domain}'.rstrip('/')
        url = f'{base_url}{api_path}'

        # Header Value for Content type + APIKEY Authentication (matches C# headers)
        self.session.headers.update({
            'Accept': 'application/json',
            'x-api-key': api_key
        })

        try:
            # Post data to create contact (matching C# property names as JSON keys)
            contact_data = {
                'contactName': contact_name,
                'emailId': email,
                'contactDisplayName': display_name
            }

            response = self.session.post(url, json=contact_data)

            # Read the response (like ReadAsStringAsync in C#)
            response_text = response.text

            # Confirm whether the response is success or not (like EnsureSuccessStatusCode)
            response.raise_for_status()

            self.stdout.write(self.style.SUCCESS('Result: ' + response_text))

        except requests.exceptions.HTTPError as ex:
            status_code = ex.response.status_code if ex.response is not None else None
            self.stderr.write('ERROR:')
            self.stderr.write(f'Status Code: {status_code}')
            self.stderr.write(f'Message: {str(ex)}')
            self.stderr.write('StackTrace:')
            self.stderr.write(traceback.format_exc())

        except Exception as ex:
            self.stderr.write('ERROR:')
            self.stderr.write('Status Code: N/A')
            self.stderr.write(f'Message: {str(ex)}')
            self.stderr.write('StackTrace:')
            self.stderr.write(traceback.format_exc())
